core/src/main/java/org/elasticsearch/action/search/SearchPhaseExecutionException.java

```diff
@@ -63,12 +61,26 @@ public SearchPhaseExecutionException(StreamInput in) throws IOException {
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeOptionalString(phaseName);
-        out.writeVInt(shardFailures == null ? 0 : shardFailures.length);
-        if (shardFailures != null) {
+        out.writeVInt(shardFailures.length);
+        for (ShardSearchFailure failure : shardFailures) {
+            failure.writeTo(out);
+        }
+    }
+
+    private static final Throwable deduplicateCause(Throwable cause, ShardSearchFailure[] shardFailures) {
+        if (shardFailures == null) {
+            throw new IllegalArgumentException("shardSearchFailures must not be null");
+        }
+        // if the cause of this exception is also the cause of one of the shard failures we don't add it
+        // to prevent duplication in stack traces rendered to the REST layer
```

Comment:
not sure if it's relevant.... what I wanted there is to add a cause always to this exception which wasn't the case. if we do that we don't need to add it if we have it as a shard failure already since we override `getCause` so it's equivalent.